whole-tale/terraform_deployment
# Change base image to Ubuntu, add Docker volume

## Problem(s)

* CoreOS/Container Linux is EOL 
* We frequently run out of local disk due to Docker image sizes
* The [proposed solution for implementing MATLAB support](https://github.com/whole-tale/whole-tale/issues/85#issuecomment-663705714) will increase the image cache requirements and requires a more recent Docker version.

Fixes #44.

## Approach

Modify the terraform deployment to use Ubuntu
* Change the base image from CoreOS to Ubuntu 20.04 LTS
* Port configuration from `config.ign` to `pre-setup.sh`
* Add external volume for Docker image cache

## How to test

1. Provision new cluster using `terraform_deployment` (i.e, configure `variables.tf` and `terraform apply`)
1. Restore database from production backup
1. Confirm deployment end-to-end
